packages/safe-apps-sdk/src/communication/index.ts

```diff
@@ -55,14 +59,42 @@ class PostMessageCommunicator implements Communicator {
     }
   };
 
-  public send = <M extends Methods, P, R>(method: M, params: P): Promise<SuccessResponse<R>> => {
+  private comparePermissions(current: Permission[], required: Methods[]): boolean {
+    return required.every((method: Methods) => {
+      return !!current.find((p) => p.parentCapability === method);
+    });
+  }
+
+  private async checkPermissions(requiredPermissions: Methods[]): Promise<boolean> {
+    let currentPermissions = await this.wallet.getPermissions();
+    if (!this.comparePermissions(currentPermissions, requiredPermissions)) {
+      currentPermissions = await this.wallet.requestPermissions(requiredPermissions.map((p) => ({ [p]: {} })));
+    }
+
+    return this.comparePermissions(currentPermissions, requiredPermissions);
+  }
+
+  public send = async <M extends Methods, P, R>(
+    method: M,
+    params: P,
+    requiredPermissions?: Methods[],
+  ): Promise<SuccessResponse<R>> => {
     const request = MessageFormatter.makeRequest(method, params);
 
+    if (Array.isArray(requiredPermissions)) {
+      const hasPermissions = await this.checkPermissions(requiredPermissions);
+
+      if (!hasPermissions) {
+        throw new PermissionsError('Permissions rejected', PERMISSIONS_REQUEST_REJECTED);
+      }
+    }
```

Comment:
Why do we need this logic in the SDK? I think all permission-related checking should be done on the interface side

Reply:
The sdk shouldn't tell if we have permissions or not. The interface should, SDK is only responsible for preparing messages, sending them, and receiving responses